pandas/tseries/tools.py

```diff
@@ -483,6 +484,130 @@ def parse_time_string(arg, freq=None, dayfirst=None, yearfirst=None):
 normalize_date = tslib.normalize_date
 
 
+# Fixed time formats for time parsing
+_time_formats = ["%H:%M", "%H%M", "%I:%M%p", "%I%M%p", 
+                 "%H:%M:%S", "%H%M%S", "%I:%M:%S%p", "%I%M%S%p"]
+
+
+def _guess_time_format_for_array(arr):
+    # Try to guess the format based on the first non-NaN element
+    non_nan_elements = com.notnull(arr).nonzero()[0]
+    if len(non_nan_elements):
+        element = arr[non_nan_elements[0]]
+        for time_format in _time_formats:
+            try:
+                datetime.strptime(element, time_format)
+                return time_format
+            except:
+                continue
```

Comment:
use ``pass`` rather than ``continue``. can you list the exceptions you want to catch (``ValueError``, maybe ``TypeError``)?